src/components/landing/pages/history/history-toolbar/import-history-button.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useRef, useState } from 'react'
+import { Button, Modal } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { ForkAwesomeIcon } from '../../../../../fork-awesome/fork-awesome-icon'
+import { HistoryEntry } from '../history'
+
+export interface ImportHistoryButtonProps {
+  onImportHistory: (entries: HistoryEntry[]) => void
+}
+
+export const ImportHistoryButton: React.FC<ImportHistoryButtonProps> = ({ onImportHistory }) => {
+  const { t } = useTranslation()
+  const uploadInput = useRef<HTMLInputElement>(null)
+  const [show, setShow] = useState(false)
+  const [fileName, setFilename] = useState('')
```

Comment:
You're using the filename only in the error modal. How about replacing the `show` state with a null check on `filename` ?

Reply:
That could be done

Reply:
ok